### PR TITLE
Added test to verify negative result of clSetKernelArg with CL_INVALID_ARG_INDEX

### PR DESCRIPTION
**Pull request by shajder**
Related to #2282, according to work plan from [here](https://github.com/KhronosGroup/OpenCL-CTS/issues/2282#issuecomment-3069182773)

### CODE REVIEW

File: test_conformance/api/test_kernels.cpp [hunk 1]
```diff
@@ -704,3 +704,28 @@ REGISTER_TEST(negative_set_immutable_memory_to_writeable_kernel_arg)
...
+
+    // Setup the test
+    error = create_single_kernel_helper(context, &program, nullptr, 1,
+                                        sample_single_test_kernel, nullptr);
+    test_error(error, "Unable to build test program");
+
+    sampler_arg_kernel = clCreateKernel(program, "sample_test", &error);
+    test_error(error, "Unable to get sample_test kernel for built program");
+
+    // Run the test - CL_INVALID_ARG_INDEX
```
Comment (EwanC) [inline test_conformance/api/test_kernels.cpp]: This comment doesn't give the reader much extra info as-is, it would add value if instead the comment said that the "sample_test" test kernel has 2 arguments, so `2` is out of range as arguments are indexed from zero.

File: test_conformance/api/test_kernels.cpp [hunk 2]
```diff
@@ -704,3 +704,28 @@ REGISTER_TEST(negative_set_immutable_memory_to_writeable_kernel_arg)
...
+    error = clSetKernelArg(sampler_arg_kernel, 2, sizeof(cl_sampler), nullptr);
```
Comment (EwanC) [inline test_conformance/api/test_kernels.cpp]: `cl_sampler` types aren't used anywhere else in this file, can we use `sizeof(cl_mem)` to avoid a reader having to work if there's a reason we've started using `cl_sampler` for this particular test.

File: test_conformance/api/test_kernels.cpp [hunk 3]
```diff
@@ -704,3 +704,28 @@ REGISTER_TEST(negative_set_immutable_memory_to_writeable_kernel_arg)
 
     return TEST_PASS;
 }
+
+REGISTER_TEST(negative_invalid_arg_index)
+{
+    cl_int error = CL_SUCCESS;
+    clProgramWrapper program;
+    clKernelWrapper sampler_arg_kernel;
```
Comment (EwanC) [inline test_conformance/api/test_kernels.cpp]: The kernel names at the top of the file are "sample" rather than "sampler", a "sampler" is a separate concept to what we're trying to test here, so would prefer we didn't complicate the test by using that terminology. Would just call this variable `kernel`.

**Review by EwanC — APPROVED** [review]
Some nitpicks but LGTM

File: test_conformance/api/test_kernels.cpp (hunk 1 above)
Comment (shajder) [reply test_conformance/api/test_kernels.cpp]: Corrected

File: test_conformance/api/test_kernels.cpp (hunk 2 above)
Comment (shajder) [reply test_conformance/api/test_kernels.cpp]: Done

File: test_conformance/api/test_kernels.cpp (hunk 3 above)
Comment (shajder) [reply test_conformance/api/test_kernels.cpp]: Done :)

**Review by bashbaug — APPROVED** [review]
Merging as discussed in the August 12th teleconference.
